samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -64,21 +65,25 @@
   private String newJobModelVersion;  // version published in ZK (by the leader)
   private JobModel jobModel;
 
-  public ZkJobCoordinator(String processorId, String groupId, Config config, ScheduleAfterDebounceTime debounceTimer, ZkUtils zkUtils,
+  public ZkJobCoordinator(String groupId, Config config, ScheduleAfterDebounceTime debounceTimer,
```

Comment:
> GroupId - is probably the future applicationId,

We should figure out what the semantics of groupId are and how it should be wired in from the user-facing API. 

> I will generate it right here and later will pass it from the ApplicationRunner when it is available.

Please don't generate it here if it is expected to be provided by some other component. ApplicationRunner is available now.